Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -853,7 +854,7 @@ static void webkitWebViewBaseDispose(GObject* gobject)
         webkitWebViewAccessibleSetWebView(WEBKIT_WEB_VIEW_ACCESSIBLE(webView->priv->accessible.get()), nullptr);
 #endif
 
-    GtkSettingsManager::singleton().removeObserver(webView);
+    SystemSettingsGLib::singleton().removeObserver(webView);
```

Comment:
I don't think it's a good idea to use the SystremSettings singleton from the UI process, since it's notifying about changes to RenderTheme, for example, that only makes sense in the web process. Here can use SystemSettings::State as the store for the UI process manager.

Reply:
How should each view get notified of the settings change?

Maybe we could just have a separate `applySettings` function only for the UI process that doesn't update external state like `RenderTheme`.

Reply:
Really i think it would be best if it didn’t call out anywhere, rather using observers everywhere needed, so always safe

Reply:
I agree, it would be better to have the system settings object not aware of other webcore classes, and use observers everywhere.